feat: lookup project tsconfig

slhck opened the pull request:
This looks up the default tsconfig.json file for the current project when used from the CLI, to avoid possible compliation errors. Fixes #2062

arthurfiorette commented:
please create a test for when no tsconfig files can be found

slhck commented:
If no tsconfig.json file is found, the behavior is exactly the same as before this commit — you can see the only change is the assignment in the arguments. Since you have no real CLI tests suite, I'm not sure how this could be easily integrated.

slhck commented:
(Or do you mean just in the unit test, because of the missing code coverage for the "not found" case?)

slhck commented:
Please check if the new tests suffice.

domoritz commented on src/Utils/findTsConfig.ts:
```diff
@@ -0,0 +1,22 @@
+ * @param filename - The filename of the tsconfig.json file. Defaults to "tsconfig.json".
+ * @returns The path to the tsconfig.json file or undefined if no such file was found.
+ */
+export function findTsConfig(directory: string = process.cwd(), filename = "tsconfig.json"): string | undefined {
```
Is there any chance that we can import this logic from tsc rather than implementing it? If not, then this is good.

arthurfiorette commented:
Maybe (to offload this logic from this codebase) https://www.npmjs.com/package/get-tsconfig?activeTab=readme could be a good idea.

domoritz commented:
Oh look at the bottom of the read me. They show how to use tsc.

slhck commented:
Since you don't depend on `tsc`, I don't think that is a viable option, unless you assume users will always have `tsc` present in their modules.

The third-party package you linked to looks like another option (there's a package for everything in Node these days, huh), but it means having another depdendency (with its own dependencies) for something rather trivial, and you don't need its other parsing features. In essence, it does [exactly the same](https://github.com/privatenumber/get-tsconfig/blob/master/src/utils/find-up.ts) as the function I committed.

Feel free to merge or just replace the one function call with the third-party dependency — I am not passionate about this.

domoritz commented:
We depend on typescript. 

And yes I agree, no to another dep.

slhck commented:
> We depend on typescript.

Where?

https://github.com/vega/ts-json-schema-generator/blob/46f638ea799cde3d44b113a95bb307c64f2f6021/package.json#L58-L95

slhck commented:
Oh, never mind, I was looking in the wrong place. Should I update the PR or could you make that two line change yourself?

domoritz commented:
Go ahead!

arthurfiorette commented:
> Oh look at the bottom of the read me. They show how to use tsc.

How tf i have never seen this 🤡

domoritz commented:
Can you refactor the last bit @slhck?

slhck committed:
feat: lookup project tsconfig

This looks up the default tsconfig.json file for the current project when used from the CLI, to avoid possible compliation errors. Fixes #2062

slhck commented:
Please see the new commit.

swift502 commented:
I got burned on this last week, happy to see it getting improved. 👏

domoritz commented:
Thank you!
